src/forms/renderers/Overrides/material/complex/MaterialOneOfRenderer_Format.tsx

```diff
@@ -0,0 +1,207 @@
+            (el: any) => el !== null
+        );
+        return renderer;
+    });
+
+    const openNewTab = (newIndex: number) => {
+        const tabSchema = oneOfRenderInfos[newIndex].schema;
+        const { properties: tabSchemaProps } = tabSchema;
+        const defaultVal: {
+            formatter: string;
+            settings?: any;
+        } = {
+            formatter: tabSchemaProps?.formatter.default ?? '',
+        };
+
+        if (has(tabSchemaProps, 'settings')) {
+            defaultVal.settings = {};
+        }
+
+        handleChange(path, defaultVal);
+        setSelectedIndex(newIndex);
+    };
+
+    const confirm = useCallback(() => {
+        handleChange(path, null);
+        openNewTab(newSelectedIndex);
+        setOpen(false);
+    }, [handleChange, createDefaultValue, newSelectedIndex]);
+
+    const handleTabChange = useCallback(
+        (_event: any, newOneOfIndex: number) => {
+            setNewSelectedIndex(newOneOfIndex);
+            if (isEmpty(data) || !has(data, 'settings')) {
+                openNewTab(newOneOfIndex);
+            } else {
+                setOpen(true);
+            }
+        },
+        [setOpen, setSelectedIndex, data]
+    );
+
+    return (
+        <Hidden xsUp={!visible}>
+            <CombinatorProperties
+                schema={schema}
+                combinatorKeyword="oneOf"
+                path={path}
+            />
+            <Tabs value={selectedIndex} onChange={handleTabChange}>
+                {oneOfRenderInfos.map((oneOfRenderInfo) => (
+                    <Tab
+                        key={oneOfRenderInfo.label}
+                        label={oneOfRenderInfo.label}
+                    />
+                ))}
+            </Tabs>
+            {oneOfRenderInfos.map(
+                (oneOfRenderInfo, oneOfIndex) =>
+                    selectedIndex === oneOfIndex && (
+                        <JsonFormsDispatch
+                            key={oneOfIndex}
+                            schema={oneOfRenderInfo.schema}
+                            uischema={oneOfRenderInfo.uischema}
+                            path={path}
+                            renderers={renderers}
+                            cells={cells}
+                        />
+                    )
+            )}
+            <Dialog
+                open={open}
+                onClose={handleClose}
+                aria-labelledby="alert-dialog-title"
+                aria-describedby="alert-dialog-description"
+            >
+                <DialogTitle id="alert-dialog-title">
+                    {'Clear form?'}
+                </DialogTitle>
+                <DialogContent>
+                    <DialogContentText id="alert-dialog-description">
+                        Your data will be cleared if you navigate away from this
+                        tab. Do you want to proceed?
+                    </DialogContentText>
+                </DialogContent>
+                <DialogActions>
```

Comment:
This modal serves a similar purpose as the _Are you sure?_ modal, which has _continue_ and _cancel_ CTAs. Why opt for _yes_ and _no_ CTAs in this case?

Reply:
To match JSON Forms.